src/qbox_statemachine_demo.py

```diff
@@ -0,0 +1,219 @@
+class QboxConfig():
+    SUCCESS_200 = 1
+    SUCCESS_PAYLOAD = 2
+    FAIL_404 = 3
+    FAIL_REST = 4
+    SENDBACK_200 = 5
+    SENDBACK_PAYLOAD = 6
+    SENDBACK_404 = 7
+    DoNothing = 8
+
+config = {
+    "appname": "frontend",
+    "onrequest": {
+        "requestname": "sendreq",
+        "transactions": [
+            {
+                "dstname": "nodejs",
+                "request": "post_nodejs",
+                "success": QboxConfig.SUCCESS_200,
+                "fail": QboxConfig.FAIL_REST,
+                "compensation": "depost_nodejs",
+                
+                "timeout": 30
+            },
+            {
+                "dstname": "ruby",
+                "request": "post_ruby",
+                "success": QboxConfig.SUCCESS_200,
+                "fail": QboxConfig.FAIL_REST,
+                "compensation": "depost_ruby",
+                "timeout": 30
+            },
+            {
+                "dstname": "ruby2",
+                "request": "post_ruby2",
+                "success": QboxConfig.SUCCESS_200,
+                "fail": QboxConfig.FAIL_REST,
+                "compensation": "depost_ruby2",
+                "timeout": 30
+            },
+        ],
+        "onSuccess": QboxConfig.SENDBACK_200,
+        "onFail": QboxConfig.SENDBACK_404,
+    }
+}
+
+def send_requests(config):
+    print("requests sent")
+    return 1
+
+test1 = [("nodejs", QboxConfig.FAIL_404), ("ruby", QboxConfig.SUCCESS_200), ("ruby2", QboxConfig.FAIL_404)]
+test_index = 0
+def get_respond():
+    global test_index
+    res = test1[test_index]
+    test_index += 1
+    return res
+
+def send_compensation(config):
+    print("compensation sent")
+    return 1
+
+def send_back_result_to_requestor(config, status):
+    if status:
+        s = "failed"
+    else:
+        s = "succeed"
+    print("result is sent back to requestor, this time it is " + s)
+    return 1
+
+class StateInfo:

```

Comment:
Python best practice: all classes should be new-style classes i.e. they should inherit from `object`. This gives them a series of nice features, including serialization and marshalling support.